camel/models/model_manager.py

```diff
@@ -0,0 +1,220 @@
+        self.current_model = self.models[0]
+
+        # Set the scheduling strategy; default is round-robin
+        try:
+            self.scheduling_strategy = getattr(self, scheduling_strategy)
+        except AttributeError:
+            logger.warning(
+                f"Provided strategy: {scheduling_strategy} is not implemented."
+                f"Using default 'round robin'"
+            )
+            self.scheduling_strategy = self.round_robin
+
+    @property
+    def model_type(self) -> UnifiedModelType:
+        r"""Return type of the current model.
+        Returns:
+            Union[ModelType, str]: Current model type.
+        """
+        return self.current_model.model_type
+
+    @property
+    def model_config_dict(self) -> Dict[str, Any]:
+        r"""Return model_config_dict of the current model.
+        Returns:
+            Dict[str, Any]: Config dictionary of the current
+            model.
+        """
+        return self.current_model.model_config_dict
+
+    @model_config_dict.setter
+    def model_config_dict(self, model_config_dict: Dict[str, Any]):
+        r"""Set model_config_dict to the current model.
+        Args:
+            model_config_dict (Dict[str, Any]): Config dictionary
+            to be set at current model.
+        """
+        self.current_model.model_config_dict = model_config_dict
+
+    @property
+    def current_model_index(self) -> Optional[int]:
+        r"""Return the index of current model in self.models list.
+        Returns:
+            int: index of current model in given list of models.
+        """
+        return self.models.index(self.current_model)
+
+    @property
+    def token_limit(self):
+        r"""Returns the maximum token limit for a given model.
+
+        This method retrieves the maximum token limit either from the
+        `model_config_dict` or from the model's default token limit.
+
+        Returns:
+            int: The maximum token limit for the given model.
+        """
+        return self.current_model.token_limit
+
+    @property
+    def token_counter(self) -> BaseTokenCounter:
+        r"""Return token_counter of the current model.
+        Returns:
+            BaseTokenCounter The token counter following the model's
+                tokenization style.
+        """
+        return self.current_model.token_counter
+
+    def add_strategy(self, name: str, strategy_fn: Callable):
+        r"""Add a scheduling strategy method provided by user in case when none
+            of existent strategies fits.
+            When custom strategy is provided, it will be set as
+            "self.scheduling_strategy" attribute.
+
+        Args:
+            name (str): The name of the strategy.
+            strategy_fn (Callable): The scheduling strategy function.
+        """
+        if not callable(strategy_fn):
+            raise ValueError("strategy_fn must be a callable function.")
+        setattr(self, name, strategy_fn)
```

Comment:
seems one bug here, otherwise the example code would not able to run
```suggestion
        setattr(self, name, strategy_fn.__get__(self))
```